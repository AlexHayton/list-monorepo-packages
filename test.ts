/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable @typescript-eslint/no-unsafe-call */
import path, { dirname } from 'node:path';
import { fileURLToPath } from 'node:url';
import getPackages from '.';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const LERNA_DIR = path.join('./fixture/lerna');
const YARN_DIR_1 = path.join('./fixture/yarn');
const YARN_DIR_2 = path.join('./fixture/yarn-2');
const BOLT_DIR = path.join('./fixture/bolt');
const PNPM_DIR = path.join('./fixture/pnpm');

describe('getPackages()', () => {
  test('lerna matches snapshot', async () => {
    expect(await getPackages(LERNA_DIR)).toMatchSnapshot();
  });

  test('yarn matches snapshot', async () => {
    expect(await getPackages(YARN_DIR_1)).toMatchSnapshot();
  });

  test('yarn 1.4.2+ matches snapshot', async () => {
    expect(await getPackages(YARN_DIR_2)).toMatchSnapshot();
  });

  test('bolt matches snapshot', async () => {
    expect(await getPackages(BOLT_DIR)).toMatchSnapshot();
  });

  test('pnpm matches snapshot', async () => {
    expect(await getPackages(PNPM_DIR)).toMatchSnapshot();
  });

  test('returns empty array when no packages', async () => {
    expect(await getPackages(__dirname)).toEqual([]);
  });
});
