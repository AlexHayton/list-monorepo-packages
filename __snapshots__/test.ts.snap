// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPackages() bolt matches snapshot 1`] = `
[
  {
    "location": "fixture/bolt/packages/bar",
    "package": {
      "name": "bar-workspace-package",
      "version": "1.0.0",
    },
  },
  {
    "location": "fixture/bolt/packages/foo",
    "package": {
      "name": "foo-workspace-package",
      "version": "1.0.0",
    },
  },
]
`;

exports[`getPackages() lerna matches snapshot 1`] = `
[
  {
    "location": "fixture/lerna/packages/bar",
    "package": {
      "name": "bar",
    },
  },
  {
    "location": "fixture/lerna/packages/foo",
    "package": {
      "name": "foo",
    },
  },
]
`;

exports[`getPackages() pnpm matches snapshot 1`] = `
[
  {
    "location": "fixture/pnpm/packages/bar",
    "package": {
      "name": "bar",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/pnpm/packages/foo",
    "package": {
      "name": "foo",
      "version": "0.0.0",
    },
  },
]
`;

exports[`getPackages() yarn 1.4.2+ matches snapshot 1`] = `
[
  {
    "location": "fixture/yarn-2/packages/bar",
    "package": {
      "name": "bar",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/yarn-2/packages/foo",
    "package": {
      "name": "foo",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/yarn-2/package-2",
    "package": {
      "name": "package-2",
      "version": "0.0.0",
    },
  },
]
`;

exports[`getPackages() yarn matches snapshot 1`] = `
[
  {
    "location": "fixture/yarn/packages/bar",
    "package": {
      "name": "bar",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/yarn/packages/foo",
    "package": {
      "name": "foo",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/yarn/packages/with-node_modules",
    "package": {
      "name": "with-node_modules",
      "version": "0.0.0",
    },
  },
  {
    "location": "fixture/yarn/package-2",
    "package": {
      "name": "package-2",
      "version": "0.0.0",
    },
  },
]
`;
